llvm/lib/YkIR/YkIRWriter.cpp

```diff
@@ -6,30 +6,120 @@
 
 #include "llvm/ADT/Triple.h"
 #include "llvm/BinaryFormat/ELF.h"
+#include "llvm/IR/Constant.h"
 #include "llvm/IR/Constants.h"
+#include "llvm/IR/InstrTypes.h"
+#include "llvm/IR/Instructions.h"
 #include "llvm/IR/Module.h"
 #include "llvm/MC/MCContext.h"
 #include "llvm/MC/MCSectionELF.h"
 #include "llvm/MC/MCStreamer.h"
+#include "llvm/Support/ErrorHandling.h"
 
 using namespace llvm;
 using namespace std;
 
 namespace {
 
+class SerialiseInstructionException {
+private:
+  string S;
+
+public:
+  SerialiseInstructionException(string S) : S(S) {}
+  string &what() { return S; }
+};
+
 const char *SectionName = ".yk_ir";
 const uint32_t Magic = 0xedd5f00d;
 const uint32_t Version = 0;
 
 enum OpCode {
   Nop = 0,
+  Load,
+  Store,
+  Alloca,
+  Call,
+  GetElementPtr,
+  Branch,
+  ICmp,
+  BinaryOperator,
+  Ret,
+  UnimplementedInstruction = 255, // Will eventually be deleted.
```

Comment:
`YKFIXME`?

Reply:
Fixed in 498ed84d9c719357dbd5acfa201ec5ebec814e26